azurelinuxagent/common/protocol/wire.py

```diff
@@ -1627,6 +1622,13 @@ def parse_plugin_settings(self, ext_handler, plugin_settings):
             logger.error("Invalid extension settings")
             return
 
+        depends_on_node = find(settings[0], "DependsOn")
+
+        try:
+            depends_on_level = int(getattrib(depends_on_node, "dependencyLevel"))
+        except (ValueError, TypeError):
+            depends_on_level = 0
```

Comment:
do we need to catch these errors? this bumps the priority to 0.

Reply:
We expect string value corresponding to an integer value, which is parsed to get the integer value.
In order to guard against any corrupted value, we catch this exception. The dependency level is set to default in in that case. We assume that that kind of extension does not depend on anything so it gets its priority bumped up. Do you have any better suggestion in this case?

Reply:
OK, I double-checked the previous code and it was doing the same (setting the priority to 0 on error). I think we should log a warning in case bumping it up to 0 affects any dependencies and causes an extension failure.

Reply:
I think this warning is going to create a lot of noise in the logs if the platform is only sending down dependencyLevel for the cases where dependencies exist.

@sirajudeens are we sending down dependencyLevel all the time regardless or only when the customer specifies dependencies? What about for single VMs?